frame/staking-rewards/src/lib.rs

```diff
@@ -356,7 +392,13 @@ pub mod pallet {
 			// Only the owner is able to select an arbitrary `to` account.
 			let nft_owner = T::get_protocol_nft_owner::<StakingNFTOf<T>>(&instance_id)?;
 			let to = if nft_owner == who { to } else { nft_owner };
-			<Self as Staking>::claim(&instance_id, &to)?;
+			let (asset, compound) = <Self as Staking>::claim(&instance_id, &to)?;
+			if compound > T::Balance::zero() {
+				RunningTotalShares::<T>::try_mutate(asset, |total_shares| -> DispatchResult {
+					*total_shares = total_shares.safe_sub(&compound.into())?;
+					Ok(())
+				})?;
+			}
```

Comment:
looks like, there should be:
* a error if `compound == T::Balance::zero() `
* a event before `Ok(())`